src/Pagination/PaginateDirective.php

```diff
@@ -126,18 +126,33 @@ public function resolveField(FieldValue $fieldValue): FieldValue
                     $this->directiveArgValue('scopes') ?? []
                 );
 
-            return PaginationArgs::extractArgs($args, $this->paginationType(), $this->paginateMaxCount())
+            return PaginationArgs::extractArgs($args, $this->paginationType($resolveInfo), $this->paginateMaxCount())
                 ->applyToBuilder($query);
         });
 
         return $fieldValue;
     }
 
-    protected function paginationType(): PaginationType
+    protected function paginationType(?ResolveInfo $resolveInfo = null): PaginationType
```

Comment:
When is `ResolveInfo` not given?

Reply:
When we build the schema, there we also need the type to correctly add the types but we obviously don't have resolve info nor do we want it there.